ocaml/xapi/xapi_clustering.mli

```diff
@@ -0,0 +1,93 @@
+(* Copyright (C) Cloud Software Group Inc.
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation; version 2.1 only. with the special
+   exception on linking described in file LICENSE.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+*)
+
+val set_ha_cluster_stack : __context:Context.t -> unit
+
+val with_clustering_lock : string -> (unit -> 'a) -> 'a
+
+val pif_of_host :
+  __context:Context.t -> API.ref_network -> API.ref_host -> 'a Ref.t * API.pIF_t
+
+val ip_of_pif : 'a Ref.t * API.pIF_t -> Cluster_interface.address
+
+val assert_pif_prerequisites : 'a Ref.t * API.pIF_t -> unit
+
+val assert_pif_attached_to :
+  __context:Context.t -> host:[`host] Ref.t -> pIF:[`PIF] Ref.t -> unit
+
+val handle_error : Cluster_interface.error -> 'a
+
+val assert_cluster_host_can_be_created :
+  __context:Context.t -> host:'a Ref.t -> unit
+
+val get_required_cluster_stacks :
+  __context:Context.t -> sr_sm_type:string -> string list
+
+val assert_cluster_stack_valid : cluster_stack:string -> unit
+
+val with_clustering_lock_if_needed :
+  __context:Context.t -> sr_sm_type:string -> string -> (unit -> 'a) -> 'a
+
+val with_clustering_lock_if_cluster_exists :
+  __context:Context.t -> string -> (unit -> 'a) -> 'a
+
+val find_cluster_host :
+  __context:Context.t -> host:[`host] Ref.t -> 'a Ref.t option
+
+val get_network_internal :
+  __context:Context.t -> self:[`Cluster] Ref.t -> [`network] Ref.t
+
+val assert_cluster_host_enabled :
```

Comment:
Are these assertions used outside the implementation?

Reply:
I removed all those that are not used outside this module, by going through them one by one and see if it compiles